src/runnable.coffee

```diff
@@ -10,6 +10,8 @@ define ['settings'], (Settings) ->
     then discards it.
     ###
     @run: ->
+      return @ unless Settings.window[Settings.command_queue_name].q
```

Comment:
You can keep the queue in a variable and replace all references in the `run` method

```suggestion
      queue = Settings.window[Settings.command_queue_name]?.q

      return @ unless queue
```

Reply:
+1 strongly agree for something like:

```coffee
@run: ->
  queue = Settings.window[Settings.command_queue_name].q
  return @ unless queue

  for _, cmd of queue.slice()
    [category, type, args...] = cmd

    if @_commands[category]?[type]?
      @_commands[category][type].apply(@, args)
      queue.splice(queue.indexOf(cmd), 1)

  return @
```

Reply:
Done!